### PR TITLE
For the CG model, we identify and remive lipid at the same time we did for the ion and solvant

### PR DESCRIPTION
**Pull request by KarinDuong**
Now it's only available for the CG model, for AA model we keep the last one (we identify lipid at the same time we did for the protein)

### CODE REVIEW

**Comment by pierrepo** [comment]
@KarinDuong could you instead create an independent `count_remove_lipid()` function that targets lipids only. This function will be called only at the CG resolution.

**Comment by KarinDuong** [comment]
Ok for merge

File: grodecoder.py [hunk 1]
```diff
@@ -862,6 +862,70 @@ def find_ion_solvant(
     return (universe, counts)
 
 
+def count_remove_ion_solvant(
```
Comment (pierrepo) [inline grodecoder.py:865]: This function looks redundant with `find_ion_solvant()`
Could you have a look @KarinDuong ?

**Comment by pierrepo** [comment]
@KarinDuong I fixed the conflict, but the end of the function `count_remove_lipid()` does not look good. Could you please have a look?

**Comment by KarinDuong** [comment]
Seems good to me (get the good result)

File: grodecoder.py [hunk 2]
```diff
@@ -923,6 +922,142 @@ def count_remove_ion_solvant(
...
+    
+    for lipid in lipid_MAD.values:
+        universe, counts = find_lipids(lipid.tolist(), universe, counts)
+
+    # Write the new universe without ions and solvant into a new file
+    output_file = f"{Path(input_filepath).stem}_without_H_ions_solvant{Path(input_filepath).suffix}"
+    universe.atoms.write(output_file, reindex=False)
+    logger.debug(
+        f" New structure file without hydrogens, ions and solvants, lipids : {output_file}"
+    )
```
Comment (pierrepo) [inline grodecoder.py:1044]: Why don't need to save this structure.

File: grodecoder.py [hunk 3]
```diff
@@ -923,6 +922,142 @@ def count_remove_ion_solvant(
...
+
+    universe_clean = mda.Universe(output_file)
+
+    # Print which ion and solvant we find, and how many
```
Comment (pierrepo) [inline grodecoder.py:1048]: Why do we mention ions and solvant?

File: grodecoder.py (hunk 2 above)
Comment (KarinDuong) [reply grodecoder.py:1044]: Because we remove it from the system (after we counted it), and avoid to count it in the get_atom_pair

Comment (pierrepo) [reply grodecoder.py:1044]: Yes why saving the structure on disk? It's not useful.

**Comment by pierrepo** [comment]
@KarinDuong are we ready to merge?

**Comment by KarinDuong** [comment]
Ready for merge

**Pull request merged**